code/linear_model.py

```diff
@@ -3,12 +3,13 @@
 import numpy.linalg as npl
```

Comment:
This is a really long module. It would be better if we can add more documentation in it to explain the functionality of each chunk. And is it possible and necessary to break it down?